### DISCUSSION TITLE
Using Github Action environment variables in shell script

### DISCUSSION THREAD

**Discussion by empire-jlee** [community/community#27027] [Actions]
<p>Currently I’m running into issues trying to get ${{ secrets.ENVIRONMENT_VARIABLE_NAME }} work within a bash script file. Does anyone know if using ${{ &lt;context&gt; }} is allowed in a bash script file? I can’t find any information on this in the Github Actions documentation</p>
<p>Step sample</p>
<pre><code>- name: Run migrations
  run: sh ./github/scripts/migrate.sh
</code></pre>
<p>Within migrate.sh</p>
<pre><code>echo 'HELLO_WORLD='${{ secrets.HELLO_WORLD }} &gt;&gt; .env // Code line 1
</code></pre>
<p>Git log error</p>
<pre><code>.github/scripts/migrate.sh: 1: .github/scripts/migrate.sh: Bad substitution  
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>That won’t work, because “${{ &lt;context&gt; }}” and the “secrets” variable are GitHub Actions constructs that Bash doesn’t understand. You’ll have to pass the secret to your step as an environment variable:</p>
<pre><code>- name: Run migrations
  run: sh ./github/scripts/migrate.sh
  env:
    ENVIRONMENT_VARIABLE_NAME: ${{ secrets.ENVIRONMENT_VARIABLE_NAME }}
</code></pre>
<p>And then use that environment variable in your script.</p>

**Comment by empire-jlee** [comment]
<p>I can see this solution being a great work around for those only need to pass in a couple of environment variables into their bash script file. I’ll mark this as the solution since this may benefit others in the future. Thanks!</p>

**Comment by Godtide** [comment]
<p>The below works for me</p>
<pre><code class="lang-auto">bash -eux  .githib/scripts/file.sh 
</code></pre>
<p>let more here <a href="https://github.com/guitarrapc/githubactions-lab" rel="noopener nofollow ugc">git actions lab</a></p>
